step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesBucketingHandler.java

```diff
@@ -66,6 +67,22 @@ public void processGauges(List<Measurement> measurements) {
         });
     }
 
+    /**
+     * This method will handle existing measurements, and will check if it is a gauge or normal measurement
+     * @param measurement
+     */
+    public void ingestExistingMeasurement(Measurement measurement) {
+        if (measurement == null) {
+            return;
+        }
+        measurement.remove("_id"); // because these measurements come with a generated id and can't be grouped into buckets.
+        if (measurement.getType().equals(THREAD_GROUP_MEASUREMENT_TYPE)) {
```

Comment:
Good, that you caught this special case. We might have to refactor the handling of gauges in the future in order not to have to treat them as special case...